modules/nlu/src/backend/language-provider.ts

```diff
@@ -110,20 +208,83 @@ export class RemoteLanguageProvider implements LanguageProvider {
     throw new Error(`No provider could successfully fullfil request "${path}" for lang "${lang}"`)
   }
 
-  async vectorize(tokens: string[], lang: string): Promise<number[][]> {
+  /**
+   * Generates words that don't exist in the vocabulary, but that are built from ngrams of existing vocabulary
+   * @param subsetVocab The tokens to which you want similar tokens to
+   */
+  async generateSimilarJunkWords(subsetVocab: string[]): Promise<string[]> {
+    // from totalVocab compute the cachedKey the closest to what we have
+    // if 75% of the vocabulary is the same, we keep the cache we have instead of rebuilding one
+    const gramset = vocabNGram(subsetVocab)
+    let result: string[] | undefined
+
+    this._junkwordsCache.forEach((junk, vocab) => {
+      if (!result) {
+        const sim = setSimilarity(vocab, gramset)
+        if (sim >= 0.75) {
+          result = junk
+        }
+      }
+    })
+
+    if (!result) {
+      // didn't find any close gramset, let's create a new one
+      const realWords = _.uniq(subsetVocab)
+      const meanWordSize = _.meanBy(realWords, w => w.length)
+      const minJunkSize = Math.max(JUNK_TOKEN_MIN, meanWordSize / 3)
+      const maxJunkSize = Math.min(JUNK_TOKEN_MAX, meanWordSize * 1.5)
+
+      result = _.range(0, JUNK_VOCAB_SIZE).map(() =>
+        _.sampleSize(gramset, _.random(minJunkSize, maxJunkSize, false)).join('')
+      ) // randomly generated words
```

Comment:
With ES6 you can now natively create a N size array with a mapping function;
```js
Array.from({length: JUNK_VOCAB_SIZE}, () => _.sampleSize(gramset, _.random(minJunkSize, maxJunkSize, false)).join(''))
```
Not really a change to make but fun to know since you don't have to use (new Array(N)).fill before the itteration.